clang/lib/CConv/ConstraintBuilder.cpp

```diff
@@ -560,21 +513,25 @@ void ConstraintBuilderConsumer::HandleTranslationUnit(ASTContext &C) {
     else
       errs() << "Analyzing\n";
   }
-  TypeVariableBindingsMapT TypeVariableBindings;
-  GlobalVisitor GV = GlobalVisitor(&C, Info, TypeVariableBindings);
+
+
+  VariableAdderVisitor VAV = VariableAdderVisitor(&C, Info);
+  TypeVarVisitor TV = TypeVarVisitor(&C, Info);
+  GlobalVisitor GV = GlobalVisitor(&C, Info, TV);
```

Comment:
Should we rename `GlobalVisitor` to be something more meaningful?

Reply:
Any ideas? I can understand why GlobalVisitor was chosen since it looks at thing in the global scope. It's not particularly intuitive though.

Reply:
ConstraintGenVisitor ?

Reply:
So: Rename `GlobalVisitor` to `ConstraintGenVisitor` ?